File: tournaments/applesDistribution/applesDistribution.cpp
int applesDistribution(int apples, int boxCapacity, int maxResidue) {
  int result = 0;
  for (int i = 1; i <= boxCapacity; i++) {
    if (apples % i <= maxResidue) {
      result++;
    }
  }
  return result;
}

File: tournaments/applesDistribution/applesDistribution.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub applesDistribution {
    my ($apples, $boxCapacity, $maxResidue) = @_;

    my $result = 0;

    for (my $i = 1; $boxCapacity >= $i; ++$i) {
        if ($apples % $i <= $maxResidue) {
            ++$result;
        }
    }

    return $result;
}
